### PR TITLE
Report fancier

### LINKED ISSUE #240: Make Newsletter Fancier
## Description
Dan says 'make newsletter fancier [...] business people want business things'

## Required Files
- weekly_report.py

## User Story
As a subscriber to this service I need the report to give an impression of quality and control

### PR DESCRIPTION
**Pull request by shindym**
- reformatted pdf to be neater
- added wordcloud into pdf
Closes #240

### CODE REVIEW

File: weekly_report/generate_report.py [hunk 1]
```diff
@@ -3,14 +3,16 @@
 from email.mime.text import MIMEText
 from email.mime.application import MIMEApplication
 from email.mime.multipart import MIMEMultipart
-
+from datetime import datetime
```
Comment (hhm970) [inline weekly_report/generate_report.py:6]: Put `datetime` in its own block, and all `email.mime` libraries with the other 3rd party libraries

File: weekly_report/generate_report.py [hunk 2]
```diff
@@ -191,8 +206,8 @@ def average_rating(self) -> float:
     def top_three_publishers(self) -> list[str]:
         """Returns the top three game publishers of this week."""
         with self.conn.cursor() as cur:
-            cur.execute("""SELECT p.publisher_name, count(*) AS num_games FROM game as g 
-                        LEFT JOIN publisher AS p ON (g.publisher_id = p.publisher_id) 
+            cur.execute("""SELECT p.publisher_name, count(*) AS num_games FROM game as g
```
Comment (hhm970) [inline weekly_report/generate_report.py:209]: Nice SQL query with `COUNT(*)`

File: weekly_report/generate_report.py [hunk 3]
```diff
@@ -229,6 +244,37 @@ def get_website_names(self) -> list[str]:
                 website_list.append(websites[x]['website_name'])
             return website_list
 
+    def get_word_cloud(self):
```
Comment (hhm970) [inline weekly_report/generate_report.py:247]: Very chunky function at first glance; would it possible to delegate functionality across multiple functions?

File: weekly_report/generate_report.py (hunk 1 above)
Comment (shindym) [reply weekly_report/generate_report.py:6]: Did some research it is a built in library.

File: weekly_report/generate_report.py (hunk 3 above)
Comment (shindym) [reply weekly_report/generate_report.py]: Tried to address

File: weekly_report/generate_report.py [hunk 4]
```diff
@@ -229,6 +245,43 @@ def get_website_names(self) -> list[str]:
...
+          number of times it occurs in set of website ids.."""
+        with self.conn.cursor() as cur:
+            cur.execute("""SELECT t.tag_name, count(*) from tag AS t INNER JOIN game_tag_matching AS gt ON(t.tag_id=gt.tag_id) WHERE gt.game_id IN
+                        (SELECT game_id from game AS g WHERE g.release_date >= CURRENT_DATE - INTERVAL '7 days'
+                        AND g.release_date <= CURRENT_DATE AND g.website_id IN %s) GROUP BY t.tag_name ORDER BY count(*) DESC;
+                        """, (self.website_ids,))
+            tags_word_count = cur.fetchall()
+            return tags_word_count
+
+    def get_word_cloud(self):
```
Comment (hhm970) [inline weekly_report/generate_report.py:259]: Good job on dividing functionality in this function; it is much easier to follow

**Review by hhm970 — APPROVED** [review]
The code looks well-defined, coherent, and easy to follow - I'm satisfied with it!

**Pull request merged**